Configurations/10-main.conf

```diff
@@ -695,7 +695,7 @@ my %targets = (
         shared_target    => "linux-shared",
         shared_cflag     => "-fPIC",
         shared_ldflag    => sub { $disabled{pinshared} ? () : "-Wl,-znodelete" },
-        enable           => [ "afalgeng" ],
+        enable           => [ "afalgeng", "ktls" ],
```

Comment:
The intention is to not enable KTLS by default in 3.0.  KTLS would bypass the FIPS validation.

Reply:
Okay. The validation is bypassed because part of the TLS is no longer handled by openssl but the kernel / another software stack. afalg- is different because it is an engine or is crypto algorithm different?

Distro wise, wouldn't it be good to enable it by default? Is this something that is tested by the project or is it ignored due to the FIPS part?
I'm asking so I know if it is recommended to enable it in a distro build or not.

Reply:
I think this is a question best left to the individual distro.  For a distro that doesn't care about FIPS or that is doing their own FIPS validation, enabling it seems reasonable.  There is some background in #13794

Using an engine that isn't FIPS validated would also risk breaking things in the same way.

Reply:
> I'm asking so I know if it is recommended to enable it in a distro build or not.

If you are intending to distribute the FIPS provider it could be problematic to automatically enable it. KTLS essentially avoids all use of providers and defers crypto to the kernel instead. If a user has configured the FIPS provider they might reasonably expect all crypto to be done inside the FIPS boundary. But if KTLS is unexpectedly being used then this won't be the case.

Reply:
OTOH, at the moment in master ktls is disabled by default at compile time, but if compiled in then it s is enabled by default at run time. I'm currently writing a PR which will change that so that it is disabled by default at run time. With that in place then the problem mostly goes away. You can compile it in by default but it still won't be used unless explicitly configured for use at run time.